### ISSUE TITLE
Return the message in an error if it failed to be sent

### ISSUE THREAD

**Issue by bschwind** [tonarino/actor#75]
Flume has a [TrySendError](https://docs.rs/flume/0.10.14/flume/enum.TrySendError.html) which returns the message that we attempted to send.

We should surface this to our API so actors can decide what to do with the message if it failed to send.

This might create an explosion of generics so let's approach this carefully.

**Comment by strohel** [comment]
Yeah this may end up being a bit tricky in current type design of actor:

We allow infinite `Recipient<T>` -> `Recipient<U>` -> `Recipient<V>` conversions provided the messages can be converted (V -> U -> T). But we don't want to add the bound in the other direction (V -> U), as that might not be possible for user to full at all.

So our `Recipient<V>` cannot return `SomeError<V>`, and it also cannot return `SomeError<T>`, because the whole point of `Recipient` is to erase the `T` generic parameter.

It could return `Box<dyn Any>`, but that would incur the boxing costs on all failures to send (some users may send large volume of big messages that are largely dropped, and don't care about it).

Maybe there is some other way, perhaps registering some form or callback etc.
